### PR TITLE
docs(training): decide image adapter placement

### LINKED ISSUE #519: [governance] Two training-code homes: research/training/ vs python/image_adapter/ — document or consolidate
## Context

Surfaced by the post-#507 cross-section sweep (the same one that found the publish-tarball guard gaps fixed in [#518](https://github.com/p-to-q/wittgenstein/pull/518)).

The repo now has **two** parallel Python training-code locations:

1. **`research/training/`** — the home the delivery doctrine documents (`docs/research/2026-05-13-delivery-and-componentization.md`, `research/training/README.md`). Subprograms: `tokenizer/`, `adapter/`, `llm-head/`, `_shared/`. Protected by `scripts/check-no-research-imports.mjs` (no `packages/*/src` → `research/` imports).
2. **`python/image_adapter/`** — the image style-adapter trainer (small MLP: `ImageSceneSpec` → discrete latent indices), landed via the M1B adapter work ([#512](https://github.com/p-to-q/wittgenstein/pull/512) / [#515](https://github.com/p-to-q/wittgenstein/pull/515)). 10 files incl. `train.py`, `eval_all.py`, `spectrum_check.py`, its own `requirements.txt`.

This is a real inconsistency, not a bug — but it should be resolved deliberately:

- The delivery doctrine says training lives in `research/training/`. `python/image_adapter/` is a second home that doctrine doesn't mention.
- There's even semantic overlap: `research/training/adapter/` (Phase-1 MaskGIT-style L4 seed-expander, #397) **and** `python/image_adapter/` (the scene→VQ-latent MLP) are both "adapter training" — a contributor can't tell from the tree which is canonical.

## What's already handled (so this is not urgent)

- [#518](https://github.com/p-to-q/wittgenstein/pull/518) extended the publish-surface guard to fence `python/` (and `polyglot-mini/`) out of the npm tarball, and fixed the dead `benchmarks/` pattern. So there's no leak risk while this is decided.
- CI Python coverage still only runs `polyglot-mini/` — neither `research/training/` nor `python/image_adapter/` has compile/test coverage in CI. `python/image_adapter/` even ships a `train_numpy.py` described as "good for CI / parity" that nothing runs.

## Decision needed (owner: maintainer + ML specialist)

1. **Home**: Is `python/image_adapter/` a recognized training surface that the delivery doctrine should explicitly name, or should it move under `research/training/image-adapter/` to keep one home?
2. **Naming**: If both stay, clarify the relationship between `research/training/adapter/` and `python/image_adapter/` so the two "adapter" dirs aren't confusing (one is the Phase-1 seed-expander scaffold, the other is the scene→latent MLP — name them so).
3. **CI**: Should `python/image_adapter/` get a minimal CI lane (at least `py_compile` + the pure-NumPy `train_numpy.py` smoke it advertises)? Right now its tests run nowhere.

## Non-goals

- Do not move or rewrite the ML training content as part of this issue — it's a governance/placement + CI-coverage decision, not a code rewrite.
- Do not block the M1B closeout ([#507](https://github.com/p-to-q/wittgenstein/issues/507)) on this — the guard fix (#518) already removed the only real risk.

Surfaced 2026-05-28 by the maintainer cross-section sweep.

### PR DESCRIPTION
**Pull request by koriyoshi2041**
Closes #519.

## Summary

- documents the #519 placement decision: keep `research/training/` as the canonical Phase-1 GPU training home and recognize `python/image_adapter/` only as the narrow scene-spec-to-VQ-latent MLP bridge
- clarifies the naming collision between `research/training/adapter/` and `python/image_adapter/` in both READMEs and the delivery doctrine
- adds a pure-NumPy CI smoke for `python/image_adapter/train_numpy.py` that generates a synthetic encoded dataset and verifies the exported `witt.image.adapter.mlp/v0.1` JSON contract
- makes `python/**` and `data/image_adapter/**` trigger the Node/Python CI path, and widens the package import guard so `packages/*/src` cannot import repo-root `python/` either

## Self-review

- This intentionally does not move training code or touch active model-training files.
- The decision is conservative: `python/image_adapter/` is documented as a narrow bridge lane, not a second generic training tree; if it grows into manifests, lab execution, or shared datasets, the doc requires a focused move/split into `research/training/image-adapter/`.
- The new test is contract-level and CPU-only. It proves the CI-advertised NumPy path exports the runtime JSON shape, but does not claim PyTorch training, real dataset quality, or lab evidence.
- The import guard and npm publish guard now agree on the boundary: training/research Python surfaces stay outside publish packages.

## Validation

- `pnpm install --frozen-lockfile --offline`
- `git diff --check`
- `python3 -m compileall python/image_adapter && python3 -m unittest discover -s python/image_adapter -p 'test_*.py' -v`
- `pnpm test:image-adapter`
- `pnpm exec prettier --check docs/research/2026-05-31-training-homes-decision.md`
- `pnpm format:check`
- `typos --config typos.toml .github/workflows/ci.yml package.json scripts/check-no-research-imports.mjs docs/research/2026-05-13-delivery-and-componentization.md docs/research/2026-05-31-training-homes-decision.md research/training/README.md python/image_adapter/README.md python/image_adapter/test_train_numpy_smoke.py`
- added-line length sanity on changed files: no added line over 120 chars
- `node scripts/check-no-research-imports.mjs`
- `pnpm lint:publish-surface`
- `pnpm format:check`
- `pnpm lint:deps && pnpm lint:publish-surface`
- `pnpm lint`
- `pnpm typecheck`
- `pnpm test`
- `pnpm test:golden`
- `pnpm reviewer-bench`

### CODE REVIEW

**Comment by koriyoshi2041** [comment]
CodeRabbit follow-up handled before merge:\n\n- Added a bounded  to the  subprocess smoke and surfaced stdout/stderr on timeout or non-zero exit.\n- Updated  so the import guard wording matches the widened  +  boundary.\n- Added an explicit NumPy-only CI dependency install for the image-adapter smoke. I intentionally did not install  because that file includes ; the CI lane is a CPU/NumPy contract smoke, not the full PyTorch trainer.\n- Fixed the  review-request workflow to skip self-review requests, and removed the misapplied  label from this PR because it is governance/CI placement work, not model-owner training content. The rerun  check is now green.

**Comment by koriyoshi2041** [comment]
CodeRabbit follow-up handled before merge:

- Added a bounded `timeout=120` to the `train_numpy.py` subprocess smoke and surfaced stdout/stderr on timeout or non-zero exit.
- Updated `research/training/README.md` so the import guard wording matches the widened `research/` + `python/` boundary.
- Added an explicit NumPy-only CI dependency install for the image-adapter smoke. I intentionally did not install `python/image_adapter/requirements.txt` because that file includes `torch`; the CI lane is a CPU/NumPy contract smoke, not the full PyTorch trainer.
- Fixed the `pull_request_target` review-request workflow to skip self-review requests, and removed the misapplied `needs-ml-specialist` label from this PR because it is governance/CI placement work, not model-owner training content. The rerun `request` check is now green.
